File: pms_backend/src/main/java/com/omar/isdb62/pharmacy_management_backend/model/Inventory.java
package com.omar.isdb62.pharmacy_management_backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name= "pms_inventory")
public class Inventory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //1
    @Column(name = "Company")
    private String companyName;

    //2
    @Column(name = "item_name", nullable = false)
    private String itemName;


    //3
    @Column(nullable = false)
    private String category;

    //4
    private String generic;

    //5
    @Column(nullable = false)
    private int quantity;

    //6
    @Column(name = "unit_price", precision = 10, scale = 2 , nullable = false)
    private BigDecimal unitPrice;   //regular purchase price

    //7
    @Column(name = "purchase_discount", precision = 10, scale = 2)
    private BigDecimal purchaseDiscount;

    //8
    @Column(name = "net_purchase_price", precision = 10, scale = 2)
    private BigDecimal netPurchasePrice;

    //9
    @Column(name = "sell_price", precision = 10, scale = 2, nullable = false)
    private BigDecimal sellPrice;

    //10
    @Column(name = "total_inventory_value", precision = 10, scale = 2)
    private BigDecimal totalInventoryValue;

    //11
    @Column(name = "received_date")
    private LocalDate receivedDate;  // Add this field

}

File: pms_flutter_app/lib/model/inventory.dart
class Inventory {
  final int id;
  final String companyName;
  final String itemName;
  final String category;
  final String generic;
  final int quantity;
  final double unitPrice; // Using double for BigDecimal
  final double? purchaseDiscount; // Nullable
  final double? netPurchasePrice; // Nullable
  final double sellPrice;
  final double? totalInventoryValue; // Nullable
  final String?
  receivedDate;

  Inventory({
    required this.id,
    required this.companyName,
    required this.itemName,
    required this.category,
    required this.generic,
    required this.quantity,
    required this.unitPrice,
    this.purchaseDiscount,
    this.netPurchasePrice,
    required this.sellPrice,
    this.totalInventoryValue,
    this.receivedDate,
  });

  // Factory constructor to create an Inventory object from a JSON map
  factory Inventory.fromJson(Map<String, dynamic> json) {
    return Inventory(
      id: json['id'] as int,
      // Ensure proper casting
      companyName: json['companyName'] as String,
      itemName: json['itemName'] as String,
      category: json['category'] as String,
      generic: json['generic'] as String,
      quantity: json['quantity'] as int,
      unitPrice: (json['unitPrice'] as num).toDouble(),
      // Ensure it's a double
      purchaseDiscount: (json['purchaseDiscount'] as num?)?.toDouble(),
      // Handle nullable
      netPurchasePrice: (json['netPurchasePrice'] as num?)?.toDouble(),
      // Handle nullable
      sellPrice: (json['sellPrice'] as num).toDouble(),
      // Ensure it's a double
      totalInventoryValue: (json['totalInventoryValue'] as num?)?.toDouble(),
      // Handle nullable
      receivedDate: json['receivedDate'] as String?,
    );
  }
}